src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php

```diff
@@ -729,4 +807,326 @@ public static function provideReplaceIgnoreCase()
+            ['foo', 'foo', 'foo'],
+            ['foobar', 'foobar', 'bar'],
+            ['foobar', 'foo', 'bar'],
+            ['foo', 'foofoofoo', 'foo'],
+            ['foobar', 'foobarbarbar', 'bar'],
+            ['fooBarbar', 'fooBar', 'bar'],
+            ["foo\nfoo", "foo\n", 'foo'],
+        ];
+    }
+
+    /**
+     * @dataProvider provideEnsureEndIgnoreCase
+     */
+    public function testEnsureEndIgnoreCase(string $expectedString, string $origin, $suffix)
+    {
+        $instance = static::createFromString($origin)->ignoreCase()->ensureEnd($suffix);
+
+        $this->assertEquals(static::createFromString($expectedString), $instance);
+    }
+
+    public static function provideEnsureEndIgnoreCase()
+    {
+        return [
+            ['', '', ''],
+            ['foo', 'foo', ''],
+            ['foo', '', 'foo'],
+            ['foo', 'foo', 'foo'],
+            ['fooBar', 'fooBar', 'bar'],
+            ['foobar', 'foo', 'bar'],
+            ['fOo', 'fOofoOFoo', 'foo'],
+            ['fooBar', 'fooBarbArbaR', 'bar'],
+            ["foo\nfoo", "foo\n", 'foo'],
+        ];
+    }
+
+    /**
+     * @dataProvider provideCollapseWhitespace
+     */
+    public function testCollapseWhitespace(string $expectedString, string $origin)
+    {
+        $instance = static::createFromString($origin)->collapseWhitespace();
+
+        $this->assertEquals(static::createFromString($expectedString), $instance);
+    }
+
+    public static function provideCollapseWhitespace()
+    {
+        return [
+            ['', ''],
+            ['', " \t\r\n"],
+            ['foo bar', 'foo bar'],
+            ['foo bar baz', ' foo   bar baz'],
+            ['foo bar baz', " foo\nbar \t baz\n"],
+        ];
+    }
+
+    /**
+     * @dataProvider provideEqualsTo
+     */
+    public function testEqualsTo(bool $expected, string $origin, $other)
+    {
+        $this->assertSame($expected, static::createFromString($origin)->equalsTo($other));
+    }
+
+    public static function provideEqualsTo()
+    {
+        return [
+            [true, '', ''],
+            [false, '', 'foo'],
+            [false, 'foo', ''],
+            [false, 'foo', 'Foo'],
+            [false, "foo\n", 'foo'],
+            [true, 'Foo bar', 'Foo bar'],
+            [true, 'Foo bar', new BinaryString('Foo bar')],
+            [true, 'Foo bar', new Utf8String('Foo bar')],
+            [true, 'Foo bar', new GraphemeString('Foo bar')],
+            [false, '', []],
+            [false, 'foo', ['bar', 'baz']],
+            [true, 'foo', ['bar', 'foo', 'baz']],
+        ];
+    }
+
+    /**
+     * @dataProvider provideEqualsToIgnoreCase
+     */
+    public function testEqualsToIgnoreCase(bool $expected, string $origin, $other)
+    {
+        $this->assertSame($expected, static::createFromString($origin)->ignoreCase()->equalsTo($other));
+    }
+
+    public static function provideEqualsToIgnoreCase()
+    {
+        return [
+            [true, '', ''],
```

Comment:
this is failing in GraphemeString

Reply:
better? (and others too?)

Reply:
this one is fixed now, but the others (camel/snake) are not.
and there are failures because of https://github.com/symfony/symfony/pull/33553#discussion_r326896082
> mb_stripos() expects parameter 3 to be int, string given